PixaView/Controller/PixaViewController.swift

```diff
@@ -0,0 +1,276 @@
+    private func portaitCollectionView(){
+        let numberOfItemsPerRow : CGFloat = 1
+        let lineSpacing:CGFloat = 5
+        let interItemSpacing :CGFloat = 5
+        let width = ( collectionView.frame.width - CGFloat((numberOfItemsPerRow)) * interItemSpacing) / numberOfItemsPerRow
+        let height = width
+        
+        collectionViewFlowLayout .sectionInsetReference = .fromSafeArea
+        collectionViewFlowLayout.collectionView?.contentMode = .scaleAspectFit
+        collectionViewFlowLayout.itemSize = CGSize (width: width, height: height)
+        collectionViewFlowLayout.sectionInset = UIEdgeInsets.zero
+        collectionViewFlowLayout.scrollDirection = .vertical
+        collectionViewFlowLayout.minimumLineSpacing = lineSpacing
+        collectionViewFlowLayout.minimumInteritemSpacing = interItemSpacing
+        collectionView.setCollectionViewLayout(collectionViewFlowLayout, animated: true)
+    }
+    
+    // Landscape orientation
+    private func landscapeLayoutCollectionView(){
+       
+        let numberOfItemsInRow: CGFloat = 2
+        let lineSpacing: CGFloat = 2
+        let interItemSpacing: CGFloat = 2
+        
+        collectionViewFlowLayout.sectionInset = UIEdgeInsets.zero
+        collectionViewFlowLayout.scrollDirection = .vertical
+        collectionViewFlowLayout.minimumLineSpacing = lineSpacing
+        collectionViewFlowLayout .sectionInsetReference = .fromSafeArea
+        collectionViewFlowLayout.minimumInteritemSpacing = interItemSpacing
+        
+        let width = ( collectionView.frame.width - CGFloat((numberOfItemsInRow - 1)) * interItemSpacing) / numberOfItemsInRow
+        let height = width
+        
+        collectionViewFlowLayout.itemSize = CGSize(width: width, height: height)
+        collectionView.setCollectionViewLayout(collectionViewFlowLayout, animated: true)
+    }
+    
+    // didSelectItemAt indexPath
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        //какие данные хотим передать
+        let cellImage = (collectionView.cellForItem(at: indexPath) as? PixaViewCell)?.pixaImageOutlet.image
+        let cellTags = (collectionView.cellForItem(at: indexPath) as? PixaViewCell)?.tagsLabelOutlet.text
+        
+        let vc = (storyboard?.instantiateViewController(withIdentifier: "DetailedViewController") as? DetailedViewController)!
+        
+        vc.tags = hitsRESULT[indexPath.row].tags
+        vc.detailedPictures = cellImage
+        vc.tags = cellTags ?? "No tags"
+        
+        self.navigationController?.pushViewController(vc, animated: true)
+    } //end of didSelectItemAt indexPath
+    
+    
+    // willDisplay cell
+    func collectionView(_ collectionView: UICollectionView, willDisplay cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
+        
+        if indexPath.row == hitsRESULT.count-1 && hitsRESULT.count != total{
+            currentPage = currentPage + 1
```

Comment:
I think we have disscussed it, we need increment current page only we it was successfully fetched fetched (inside sucsess case in fetch)